fe/fe-core/src/main/java/org/apache/doris/statistics/MetaStatisticsTask.java

```diff
@@ -17,21 +17,85 @@
 
 package org.apache.doris.statistics;
 
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.common.DdlException;
+
+import com.google.common.collect.Maps;
+
 import java.util.List;
+import java.util.Map;
 
-/*
-A statistics task that directly collects statistics by reading FE meta.
+/**
+ * A statistics task that directly collects statistics by reading FE meta.
+ * e.g. for fixed-length types such as Int type and Long type we get their size from metadata.
  */
 public class MetaStatisticsTask extends StatisticsTask {
 
-    public MetaStatisticsTask(long jobId, StatsGranularityDesc granularityDesc,
-                              StatsCategoryDesc categoryDesc, List<StatsType> statsTypeList) {
+    public MetaStatisticsTask(long jobId,
+                              StatsGranularityDesc granularityDesc,
+                              StatsCategoryDesc categoryDesc,
+                              List<StatsType> statsTypeList) {
         super(jobId, granularityDesc, categoryDesc, statsTypeList);
     }
 
     @Override
     public StatisticsTaskResult call() throws Exception {
-        // TODO
-        return null;
+        Map<StatsType, String> statsTypeToValue = Maps.newHashMap();
+        List<StatsType> statsTypeList = getStatsTypeList();
+
+        for (StatsType statsType : statsTypeList) {
+            switch (statsType) {
+                case ROW_COUNT:
+                    getRowCount(statsType, statsTypeToValue);
+                    break;
+                case DATA_SIZE:
+                    getDataSize(statsType, statsTypeToValue);
+                    break;
+                case MAX_SIZE:
+                case AVG_SIZE:
+                    getColSize(statsType, statsTypeToValue);
+                    break;
+                default:
+                    throw new DdlException("Unsupported statistics type(" + statsType + ").");
+            }
+        }
+
+        return new StatisticsTaskResult(granularityDesc, categoryDesc, statsTypeToValue);
+    }
+
+    private void getRowCount(StatsType statsType, Map<StatsType, String> statsTypeToValue) throws DdlException {
+        StatsCategoryDesc categoryDesc = getCategoryDesc();
+        long dbId = categoryDesc.getDbId();
+        long tableId = categoryDesc.getTableId();
+        Database db = Catalog.getCurrentCatalog().getDbOrDdlException(dbId);
+        OlapTable olapTable = (OlapTable) db.getTableOrDdlException(tableId);
```

Comment:
Add Preconditions.check() before this line.
Avoid situations where table id is null.

Reply:
ok